osu.Game/Overlays/Direct/PlayButton.cs

```diff
@@ -78,19 +79,50 @@ public PlayButton(BeatmapSetInfo setInfo = null)
                 loadingAnimation = new LoadingAnimation(),
             });
 
-            Playing.ValueChanged += updatePreviewTrack;
+            Playing.ValueChanged += playingStateChanged;
         }
 
+        private PreviewTrackManager previewTrackManager;
+
         [BackgroundDependencyLoader]
-        private void load(OsuColour colour, AudioManager audio)
+        private void load(OsuColour colour, PreviewTrackManager previewTrackManager)
         {
+            this.previewTrackManager = previewTrackManager;
+
             hoverColour = colour.Yellow;
-            this.audio = audio;
         }
 
         protected override bool OnClick(InputState state)
         {
-            Playing.Value = !Playing.Value;
+            if (!Playing.Value)
+            {
+                if (Preview != null)
```

Comment:
is this possible to hit? shouldn't the Playing state be reflecting correctly based on being bound to `Started` / `Stopped` events?

Reply:
It is, but it doesn't reflect whether the preview is already existing (!null) or not (null).